sdk/eventhub/azure-eventhub/tests/livetest/synctests/test_reconnect.py

```diff
@@ -64,30 +66,38 @@ def test_send_with_long_interval_sync(live_eventhub, sleep):
 @pytest.mark.liveTest
 def test_send_connection_idle_timeout_and_reconnect_sync(connstr_receivers):
     connection_str, receivers = connstr_receivers
+    # no retry, should just raise error
+    client = EventHubProducerClient.from_connection_string(conn_str=connection_str, idle_timeout=10, retry_total=0)
+    with client:
+        ed = EventData('data')
+        sender = client._create_producer(partition_id='0')
+    with sender:
+        sender._open_with_retry()
+        time.sleep(11)
+        sender._unsent_events = [ed.message]
+        with pytest.raises(error.AMQPConnectionError):
+            sender._send_event_data()
+
+    # with retry, should work
     client = EventHubProducerClient.from_connection_string(conn_str=connection_str, idle_timeout=10)
     with client:
         ed = EventData('data')
         sender = client._create_producer(partition_id='0')
     with sender:
-            sender._open_with_retry()
-            time.sleep(11)
-            sender._unsent_events = [ed.message]
-            ed.message.on_send_complete = sender._on_outcome
-            with pytest.raises((uamqp.errors.ConnectionClose,
-                                uamqp.errors.MessageHandlerError, OperationTimeoutError)):
-                # Mac may raise OperationTimeoutError or MessageHandlerError
-                sender._send_event_data()
-            sender._send_event_data_with_retry()
```

Comment:
We're no longer getting these exceptions raised because now we've built the retry logic into pyamqp rather than EH?

Reply:
well, yes and no.

as we build our retry logic into pyamqp, the `_send_event_data` would raise error when there's no retry and succeed if there's retry.

I split the test into two, one with 0 retry config client which would raise, and another with normal retry config client which would succeed